connectors/filtering/basic_rule.py

```diff
@@ -0,0 +1,167 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import datetime
+import re
+from enum import Enum
+
+from dateutil.parser import ParserError, parser
+
+from connectors.logger import logger
+
+
+def to_float(value):
+    try:
+        return float(value)
+    except ValueError:
+        return value
+
+
+def to_datetime(value):
+    try:
+        date_parser = parser()
+        parsed_date_or_datetime = date_parser.parse(timestr=value)
+
+        if isinstance(parsed_date_or_datetime, datetime.datetime):
+            return parsed_date_or_datetime
+        elif isinstance(parsed_date_or_datetime, datetime.date):
+            # adds 00:00 to the date and returns datetime
+            return datetime.datetime.combine(parsed_date_or_datetime, datetime.time.min)
+        else:
+            return value
+
+    except ParserError:
+        return value
+
+
+def to_bool(value):
```

Comment:
are you making the assumption value is a string?

if not, this function will return implicit results depending on the type.

```
>>> bool(None)
False
>>> bool(object())
True
>>> bool(list())
False
```

Reply:
Yes the value will be a string